src/coreclr/src/tools/Common/TypeSystem/Common/DefType.FieldLayout.cs

```diff
@@ -407,5 +407,11 @@ public void ComputeTypeContainsGCPointers()
 
             _fieldLayoutFlags.AddFlags(flagsToAdd);
         }
+
+        public override void EnsureFullyLoadable()
```

Comment:
Looks similar to what https://github.com/dotnet/runtime/blob/master/src/coreclr/src/tools/Common/Compiler/CompilerTypeSystemContext.Validation.cs is doing.

Placing this kind of validation into core type system is not great because everyone has a different expectation of what "loadable type" means.  Desktop CLR also won't load a type where various serialization magic methods have bad signatures, etc. The type system is already in use in non-compiler places, such as the ILVerify project, it's not just for crossgen.

We need a place where this answer can be cached because computing the whole thing is expensive. E.g. this is not checking the type implementa all interfces, and that check would be super expensive.

Crossgen2 probably only needs to validate to the extent of "won't crash the compiler" which is what the linked file above does

Reply:
Thanks for the pointer Michal. Yes this looks pretty close to the kind of validation I was looking for.

> Crossgen2 probably only needs to validate to the extent of "won't crash the compiler" which is what the linked file above does

Crossgen2 only needs minimal validation to avoid crashing the compiler, yes. Validating that interfaces are implemented is not in scope, and was even previously rejected in another PR due to complexity.